app/models/assignment.rb

```diff
@@ -153,15 +152,15 @@ def section_due_date(section)
       return due_date
     end
 
-    SectionDueDate.due_date_for(section, self)
+    SectionDueDate.due_date_for(section, self.assessment_id)
```

Comment:
Also, you can still just pass `self` here. Assignments don't have their own id since they are a type of assessment. In other words, for all assignments `self.id == self.assessment_id`. 

The only reason why there is an `assessment_id` attribute available for this object is that assignments have extra information stored in the `assignment_properties` table and `assignment_properties.assessment_id` is the foreign key that describes the association between rows in that table and the assessments table. The `Assignment` model has the following line:

```
delegate_missing_to :assignment_properties
```

which means that if rails is looking up an attribute on the Assignment model and you can't find it then rails will look for it in the AssignmentProperties model. This means that `assignment.assessment_id` is really `assignment.assignment_properties.assessment_id`.